Units/parser-typescript.r/ts-class.d/input.ts

```diff
@@ -0,0 +1,104 @@
+class CPoint {
+  x: number;
+  y: number;
+  constructor(x: number, y: number) {
+    this.x = x;
+    this.y = y;
+  }
+}
+
+class BankAccount {
+  balance = 0;
+  deposit(credit: number) {
+    this.balance += credit;
+    return this.balance;
+  }
+}
+
+class CheckingAccount extends BankAccount {
+  constructor(balance: number) {
+    super(balance);
+  }
+  writeCheck(debit: number) {
+    this.balance -= debit;
+  }
+}
+
+class List<T extends NamedItem> {
+  next: List<T> = null;
+  constructor(public item: T) {
+  }
+  insertAfter(item: T) {
+    var temp = this.next;
+    this.next = new List(item);
+    this.next.next = temp;
+  }
+  log() {
+    console.log(this.item.name);
+  }
+  // ...
+}
+
+class C {
+  x: number;
+  static x: string;
+}
+
+class Messenger {
+  message = "Hello World";
+  start() {
+    var _this = this;
+    setTimeout(function() { alert(_this.message); }, 3000);
+  }
+};
+
+class D {
+  data: string | string[];
+  getData() {
+    var data = this.data;
+    return typeof data === "string" ? data : data.join(" ");
+  }
+}
+
+class Point {
+  protected fakePointBuilder: () => { x: number, y: number };
+  constructor(public x: number, public y: number) { }
+  public length() { return Math.sqrt(this.x * this.x + this.y * this.y); }
+  static origin = new Point(0, 0);
+}
+
+class A {
+  private x: number;
+  protected y: number;
+  public fun: (a: 22 | 30, b: CPoint) => number | string;
```

Comment:
I don't think that capturing `a` and `b` would be useful at all. `fun` is like a variable for a function and its arguments are named here only to introduce types because you cannot just write argument type without argument name in typescript.  The actual function that could be set to that member doesn't need to have its arguments named like that but types must match.

Reply:
Capturing `a` and `b` from a variable type which represents a function doesn't make any sense. You can assign a function to that property that has different argument names. Typescript doesn't check it and they are here only because you cannot just put types without argument name.

Reply:
Oh, I see. I misunderstood the input line.